orttraining/orttraining/training_ops/cuda/optimizer/gradient_control_impl.cu

```diff
@@ -26,26 +58,47 @@ void InPlaceAccumulatorImpl(
     const T_GRAD* gradient,
     T* accumulated_gradient,
     size_t count) {
-  int blocksPerGrid = (int)(ceil(static_cast<float>(count) / GridDim::maxThreadsPerBlock));
+  if (count == 0)
+    return;
+
+  const int num_threads_per_block = GridDim::maxThreadsPerBlock;
   CUDA_LONG N = static_cast<CUDA_LONG>(count);
-  _InPlaceAccumulator<T, T_GRAD><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
-      gradient_buffer,
-      gradient,
-      accumulated_gradient,
-      N);
+
+  // This number is tuned among many data scales that can fit one GPU card.
+  // The data scales are also listed in unit tests.
+  const size_t large_data_scale_threshold = 4096 * 768;
```

Comment:
How do we come up with this figure? Is it empirical? Can we put in an unnamed namespace as a constexpr?

Reply:
As the comment said, "This number is tuned among many data scales that can fit one GPU card. The data scales are also listed in unit tests." I profile a set of input sizes, and the perf result  is shown in the PR desc, and pick up this one.

Reply:
Fixed the constexpr.